src/rgw/rgw_aws.cc

```diff
@@ -48,16 +48,71 @@ namespace rgw::aws {
 
 
   };
+  class AsyncCallerContext: public Aws::Client::AsyncCallerContext{
+  public:
+    reply_callback_t cb;
+    CephContext* cct;
+    AsyncCallerContext(reply_callback_t cb, CephContext* cct):cb(cb), cct(cct){}
+  };
+
+  void invoke_callback(const Aws::Lambda::LambdaClient* client,
+                              const Aws::Lambda::Model::InvokeRequest& request,
+                              const Aws::Lambda::Model::InvokeOutcome & outcome,
+                              const std::shared_ptr<const Aws::Client::AsyncCallerContext>& context){
+
+    auto derived = std::static_pointer_cast<const AsyncCallerContext>(context);
```

Comment:
why create a new shared_ptr? 
you can just call "get()" and cast to your derived type.
[nit] "derived" is not a good name

Reply:
@prateek ping

Reply:
done